cognite/extractorutils/uploader.py

```diff
@@ -560,3 +560,264 @@ def __len__(self) -> int:
+        cached until uploaded
+
+        Args:
+            rows: The rows to be inserted. Can either be a list of tuples, a list of ["rownumber": ..., "values": ...]
+                objects, a dictionary of rowNumber: data, or a SequenceData object.
+            column_external_ids: List of external id for the columns of the sequence
+            id: Sequence internal ID
+                Use if external_id is None
+            external_id: Sequence external ID
+                Us if id is None
+        """
+
+        if len(rows) == 0:
+            pass
+
+        either_id = EitherId(id=id, external_id=external_id)
+
+        if isinstance(rows, SequenceData):
+            # Already in desired format
+            pass
+        elif isinstance(rows, dict):
+            rows = [
+                {
+                    'rowNumber': row_number,
+                    'values': values
+                } for row_number, values in rows.items()
+            ]
+
+            rows = SequenceData(id=id, external_id=id, rows=rows, columns=column_external_ids)
+
+        elif isinstance(rows, list):
+            if isinstance(rows[0], tuple) or isinstance(rows[0], list):
+                rows = [
+                    {
+                        'rowNumber': row_number,
+                        'values': values
+                    } for row_number, values in rows
+                ]
+
+            rows = SequenceData(id=id, external_id=id, rows=rows, columns=column_external_ids)
+        else:
+            raise TypeError('Unsupported type for sequence rows: {}'.format(type(rows)))
+
+        with self.lock:
+            seq = self.upload_queue.get(either_id)
+            if seq is not None:
+                # Update sequence
+                seq.values = [*seq.values, *rows.values]
+                seq.row_numbers = [*seq.values, *rows.row_numbers]
+
+                self.upload_queue[either_id] = seq
+            else:
+                self.upload_queue[either_id] = rows
+
+    def upload(self) -> None:
+        """
+        Trigger an upload of the queue, clears queue afterwards
+        """
+        if len(self.upload_queue) == 0:
+            return
+
+        with self.lock:
+            for either_id, upload_this in self.upload_queue.items():
+                self._upload_single(either_id, upload_this)
+
+        try:
+            self._post_upload(upload_this)
+        except Exception as e:
+            self.logger.error("Error in upload callback: %s", str(e))
+
+        self.upload_queue.clear()
+        self.upload_queue_size = 0
+
+    @retry(
+        exceptions=CogniteAPIError,
+        tries=RETRIES,
+        delay=RETRY_DELAY,
+        max_delay=RETRY_MAX_DELAY,
+        backoff=RETRY_BACKOFF_FACTOR,
+    )
+    def _upload_single(self, either_id: EitherId, upload_this: SequenceData) -> SequenceData:
+        self._create_or_update(either_id)
```

Comment:
Calling this every time will double the amount of API requests, slowing down the extractor.

The pattern used elsewhere is to call the insert method, and create the sequence on an error. This reduces the amount of requests with quite a bit.